src/videojs-contrib-hls.js

```diff
@@ -78,6 +79,57 @@ const safeGetComputedStyle = function(el, property) {
 };
 
 /**
+ * Finds the index of the HLS playlist in a give list of playlists.
+ *
+ * @param {Object} media Playlist object to search for.
+ * @param {Object[]} playlists List of playlist objects to search through.
+ * @returns {number} The index of the playlist or -1 if not found.
+ * @function getHlsPlaylistIndex
+ */
+const getHlsPlaylistIndex = function(media, playlists) {
+  for (let i = 0; i < playlists.length; i++) {
+    let playlist = playlists[i];
+
+    if (playlist.resolvedUri === media.resolvedUri) {
+      return i;
+    }
+  }
+  return -1;
+};
+
+/**
+ * Updates the selcetedIndex of the QualityLevelList when a mediachange happens in hls.
+ *
+ * @param {QualityLevelList} qualityLevels The QualityLevelList to update.
+ * @param {PlaylistLoader} playlistLoader PlaylistLoader containing the new media info.
+ * @function handleHlsMediaChange
+ */
+const handleHlsMediaChange = function(qualityLevels, playlistLoader) {
+  let newPlaylist = playlistLoader.media();
+  let selectedIndex = getHlsPlaylistIndex(newPlaylist, playlistLoader.master.playlists);
+
+  qualityLevels.selectedIndex_ = selectedIndex;
+  qualityLevels.trigger({
+    selectedIndex,
```

Comment:
If the list is changes (e.g. a quality level is removed with `removeQualityLevel`, then it wont be the same. We do set the id of the quality level to the same as the index of its representation. With the QualityLevelList api is setup, i could guarantee this in a round about way of finding the qualitylevel by id, then get the index of that level in the underlying list. Or the quality level api could be updated for the getQualityLevelById to instead return an index.